test/rest_api/test_task_manager.py

```diff
@@ -0,0 +1,159 @@
+    check_field_correctness("keyspace", status, expected_status)
+    check_field_correctness("table", status, expected_status)
+    if "error" in expected_status:
+        assert expected_status["error"] in status["error"], "Incorrect error message"
+    else:
+        assert status["error"] == "", "Error message was set"
+
+async def wait_for_task(rest_api, task_id):
+    rest_api.send("GET", f"task_manager/wait_task/{task_id}")
+
+def assert_task_does_not_exist(rest_api, task_id):
+    resp = rest_api.send("GET", f"task_manager/task_status/{task_id}")
+    assert resp.status_code == requests.codes.internal_server_error, f"Task {task_id} is kept in memory"
+
+def check_sequence_number(rest_api, task_id, expected):
+    resp = rest_api.send("GET", f"task_manager/task_status/{task_id}")
+    resp.raise_for_status()
+    status = resp.json()
+    check_field_correctness("sequence_number", status, { "sequence_number": expected })
+
+def test_task_manager_modules(rest_api):
+    with new_test_module(rest_api):
+        resp = rest_api.send("GET", "task_manager/list_modules")
+        resp.raise_for_status()
+        modules = resp.json()
+        assert "test" in modules, "test module was not listed"
+
+def test_task_manager_tasks(rest_api):
+    with new_test_module(rest_api):
+        args0 = { "shard": 0, "keyspace": "keyspace0", "table": "table0"}
+        with new_test_task(rest_api, args0) as task0:
+            print(f"created test task {task0}")
+            args1 = { "shard": 1, "keyspace": "keyspace0", "table": "table1"}
+            with new_test_task(rest_api, args1) as task1:
+                print(f"created test task {task1}")
+                resp = rest_api.send("GET", "task_manager/list_module_tasks/test")
+                resp.raise_for_status()
+                tasks = [task0, task1]
+                for task in resp.json():
+                    task_id = task["task_id"]
+                    assert task_id in tasks, f"Unrecognized task_id={task_id}"
+                    tasks.remove(task_id)
+                assert not tasks, f"list_module_tasks did not return all tasks. remaining={tasks}"
+
+def test_task_manager_status_running(rest_api):
+    with new_test_module(rest_api):
+        args0 = { "keyspace": "keyspace0", "table": "table0"}
+        with new_test_task(rest_api, args0) as task0:
+            print(f"created test task {task0}")
+            resp = rest_api.send("GET", f"task_manager/task_status/{task0}")
+            resp.raise_for_status()
+
+            status = resp.json()
+            check_status_correctness(status, { "id": task0, "state": "running", "sequence_number": 1, "keyspace": "keyspace0", "table": "table0" })
+
+            resp = rest_api.send("GET", "task_manager/list_module_tasks/test")
+            tasks = resp.json()
+            assert tasks, "task_status unregistered task that did not finish"
+
+def test_task_manager_status_done(rest_api):
```

Comment:
I am not sure if this test should be kept. It relies on whether line 82 will finish in less than 2 seconds after the line 80.
I do not know how to test `task_status` for finished tasks without trusting that the status will be retrieved in less than ttl, though.

Reply:
Same applies to `test_task_manager_status_failed`

Reply:
Maybe add a test api to modify the task_ttl to increase it in tests that need the task to be alive for awhile after it's done, and to reduce it, to test it disappears after the ttl expires.